.github/workflows/docker_image_release_process.yml

```diff
@@ -0,0 +1,151 @@
+                  fi
+
+                  # Condition 2: If version is less than the npm version, exit with error
+                  # Note: This is technically less than or equal, but equality is checked above
+                  if [[ $(echo -e "${{ env.VERSIONS_TO_COMPARE }}" | sort -V | head -n 1) == "${{ env.PROVIDER_IMAGE_VERSION }}" ]]; then
+                    echo "Condition 2: Version bump required. Exiting with error."
+                    publish_dockerhub_provider="false"
+                    publish_dockerhub_js_bundle="false"
+                    exit 1
+                  fi
+
+                  # Condition 3: If greater than npm version, consider for Docker Hub publishing
+                  if [[ $(echo -e "${{ env.VERSIONS_TO_COMPARE }}" | sort -Vr | head -n 1) == "${{ env.PROVIDER_IMAGE_VERSION }}" ]]; then
+                    echo "Condition 3: Version bump detected. Preparing for Docker Hub publish."
+
+                    # Check if running on GitHub Actions
+                    if [[ "${{ vars.HOST_GITHUB_ACTIONS }}" == true ]]; then
+                      echo "Running on GitHub Actions. Continuing with Docker Hub publish."
+                    else
+                      echo "Running locally via act. Skipping Docker Hub publish."
+                      publish_dockerhub_provider="false"
+                      publish_dockerhub_js_bundle="false"
+                    fi
+                  fi
+
+                  # Update GitHub environment variables
+                  echo "Setting environment flags..."
+                  echo "publish_dockerhub_provider=$publish_dockerhub_provider" >> $GITHUB_ENV
+                  echo "publish_dockerhub_js_bundle=$publish_dockerhub_js_bundle" >> $GITHUB_ENV
+
+            - name: Install dependencies and build packages
+              # Install dependencies and build packages if either provider or JS bundle needs to be published
+              if: >-
+                  steps.check_provider_image_version.outputs.publish_dockerhub_provider == 'true' ||
+                  steps.check_provider_image_version.outputs.publish_dockerhub_js_bundle == 'true'
+              run: |
+                  set -e  # Exit immediately if a command exits with a non-zero status.
+                  echo "Installing dependencies..."
+                  npm ci
+                  echo "Building packages..."
+                  npm run build:all
+
+            - name: Log in to Docker Hub
+              # Only log in if either provider or JS bundle needs to be published
+              if: >-
+                  steps.check_provider_image_version.outputs.publish_dockerhub_provider == 'true' ||
+                  steps.check_provider_image_version.outputs.publish_dockerhub_js_bundle == 'true'
+              run: |
+                  echo "Logging into Docker Hub."
+                  echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin 2> /dev/null
```

Comment:
can we not do this via a token rather than someone's login? I assume this is set to yours/chris's login atm?